app/services/reports/facility_progress_service.rb

```diff
@@ -82,15 +83,88 @@ def dimension_combinations_for(indicator)
+            female: values_at(record, :daily_registrations_htn_female) - values_at(record, :daily_registrations_htn_and_dm_female),
+            transgender: values_at(record, :daily_registrations_htn_transgender) - values_at(record, :daily_registrations_htn_and_dm_transgender)
+          },
+          diabetes:
+            {all: values_at(record, :daily_registrations_dm_all) - values_at(record, :daily_registrations_htn_and_dm_all),
+             male: values_at(record, :daily_registrations_dm_male) - values_at(record, :daily_registrations_htn_and_dm_male),
+             female: values_at(record, :daily_registrations_dm_female) - values_at(record, :daily_registrations_htn_and_dm_female),
+             transgender: values_at(record, :daily_registrations_dm_transgender) - values_at(record, :daily_registrations_htn_and_dm_transgender)},
+          hypertension_and_diabetes:
+            {all: values_at(record, :daily_registrations_htn_and_dm_all),
+             male: values_at(record, :daily_registrations_htn_and_dm_male),
+             female: values_at(record, :daily_registrations_htn_and_dm_female),
+             transgender: values_at(record, :daily_registrations_htn_and_dm_transgender)}
+        }
+      end
+    end
+
+    memoize def daily_follow_ups_breakdown
+      records = Reports::FacilityDailyFollowUpAndRegistration.for_region(region).where("visit_date > ?", 30.days.ago)
+      records.each_with_object({}) do |record, hsh|
+        hsh[record.period] = {
+          hypertension: {
+            all: values_at(record, :daily_follow_ups_htn_all) - values_at(record, :daily_follow_ups_htn_and_dm_all),
+            male: values_at(record, :daily_follow_ups_htn_male) - values_at(record, :daily_follow_ups_htn_and_dm_male),
+            female: values_at(record, :daily_follow_ups_htn_female) - values_at(record, :daily_follow_ups_htn_and_dm_female),
+            transgender: values_at(record, :daily_follow_ups_htn_transgender) - values_at(record, :daily_follow_ups_htn_and_dm_transgender)
+          },
+
+          diabetes: {
+            all: values_at(record, :daily_follow_ups_dm_all) - values_at(record, :daily_follow_ups_htn_and_dm_all),
+            male: values_at(record, :daily_follow_ups_dm_male) - values_at(record, :daily_follow_ups_htn_and_dm_male),
+            female: values_at(record, :daily_follow_ups_htn_female) - values_at(record, :daily_follow_ups_htn_and_dm_female),
+            transgender: values_at(record, :daily_follow_ups_dm_transgender) - values_at(record, :daily_follow_ups_htn_and_dm_transgender)
+          },
+
+          hypertension_and_diabetes: {
+            all: values_at(record, :daily_follow_ups_htn_and_dm_all),
+            male: values_at(record, :daily_follow_ups_htn_and_dm_male),
+            female: values_at(record, :daily_follow_ups_htn_and_dm_female),
+            transgender: values_at(record, :daily_follow_ups_htn_and_dm_transgender)
```

Comment:
Considering how simple this hash is, let's explore what it would look like if the reporting views generated HTN-only and DM-only numbers, instead of HTN-all and DM-all numbers which require all the math above.

Reply:
yeah, make more sense to have the data points that we need to render views. Gonna update the view definision.